src/core/init/initial_seek_and_play.ts

```diff
@@ -15,154 +15,135 @@
-) : { seek$ : Observable<unknown>; load$ : Observable<ILoadEvents> } {
-  const seek$ = whenLoadedMetadata$(mediaElement).pipe(
-    take(1),
-    tap(() => {
-      log.info("Init: Set initial time", startTime);
-      mediaElement.currentTime = typeof startTime === "function" ? startTime() :
-                                                                   startTime;
+  /**
+   * Silent Observable (does not emit anything) which performs autoPlay if
+   * needed and which emits through `loaded$` the current loading status.
+   * Completes when done.
+   */
+  const load$ = tryBeginningPlayback(initClock$, mediaElement, autoPlay, false).pipe(
+    tap((evt) => {
+      isLoaded = true;
+      loaded$.next(evt);
     }),
-    shareReplay({ refCount: true })
-  );
+    ignoreElements());
 
-  const load$ : Observable<ILoadEvents> = seek$.pipe(
-    mergeMap(() => {
-      return canPlay(clock$, mediaElement, isDirectfile).pipe(
-        tap(() => log.info("Init: Can begin to play content")),
-        mergeMap((evt) => {
-          if (evt === "can-play") {
-            if (!mustAutoPlay) {
-              return observableOf("loaded" as const);
-            }
-            return autoPlay$(mediaElement);
-          }
-          return observableOf(evt);
-        })
-      );
-    }),
-    shareReplay({ refCount: true })
-  );
+  /**
+   * Observable trying to perform the initial seek and emitting updated clock
+   * ticks.
+   */
+  const clock$ = observableCombineLatest([initClock$, speed$]).pipe(
+    map(([tick, speed]) => {
+      /** True if the tick's position is at the right value */
+      let isRightPosition = true;
+      if (!isSeekDone) {
+        isRightPosition = tick.position === startTime;
+        isSeekDone = tryInitialSeek(tick, mediaElement, startTime);
+      }
+      const liveGap = manifest.isLive ? manifest.getMaximumPosition() - tick.position :
+                                        Infinity;
+      return {
+        position: tick.position,
+        getCurrentTime: tick.getCurrentTime,
+        duration: tick.duration,
+        isPaused: isLoaded ? tick.paused :
+                             !autoPlay,
+        liveGap,
+        readyState: tick.readyState,
+        speed,
+        stalled: tick.stalled,
+
+        // wantedTimeOffset is an offset to add to the timing's current time to have
+        // the "real" wanted position.
+        // For now, this is seen when the media element has not yet seeked to its
+        // initial position, the currentTime will most probably be 0 where the
+        // effective starting position will be _startTime_.
+        // Thus we initially set a wantedTimeOffset equal to startTime.
+        wantedTimeOffset: isRightPosition ? 0 :
+                                            startTime - tick.position,
+      };
+    }));
 
-  return { seek$, load$ };
+  return { loaded$,
+           clock$: observableMerge(load$, clock$).pipe(
```

Comment:
If the loaded$ value depends on load$ result, why should we have a loaded$ subject ?

Reply:
The idea is that it is the `clock$` observable which will perform both the initial seek and initial play (which is tightly linked to the loaded events) on subscription.

Yet loaded events are very important but also diverse: we could have a blocked autoplay, an issue with the loadedmetadata event, a working autoplay or just have loaded without autoplay.

Thus it is not easy to include that loaded information on a clock tick. As such I decided to provide two observable:
  - `clock$`: perform initial seek and autoplay at the right time on subscription and emit updated clock tick for the stream
  - loaded$`: Just a Replay which does not perform side-effects on subscription but emit the last known loaded-related events.

Basically `clock$` do the actions (and update clock tick information, e.g. to add the `wantedTimeOffset` property) and `loaded$` is only here for signalisation purposes.